Lifetimes draft

Connortsui20 opened the pull request:
I am more confident now in this, especially if we just watch leddoo's [12 min video](https://www.youtube.com/watch?v=gRAVZv7V91Q) at the end. I think that this can come before box and trait objects now.

Still working on polishing

Connortsui20 commented:
Everything up to pg 42 is ready for review

Connortsui20 commented:
Ready for review

1CoolDavid reviewed (approved):
LGTM!

jurplel commented:
**High-level feedback:**
- I think that slide 3 is worded kind of oddly. It seems like the flow of the slide jumps around. I wouldn't actually say that they are exactly like trait bounds either. Maybe this slide deserves another look?
- Slide 12, maybe remind them that the help text comes from the compile error?

Amazing work. I updated it with a new header style for the 2nd half of the semester

Connortsui20 commented:
> I think that slide 3 is worded kind of oddly. It seems like the flow of the slide jumps around. I wouldn't actually say that they are exactly like trait bounds either. Maybe this slide deserves another look?

I changed it to just be "Lifetimes are similar to trait bounds". They are actually very similar

I also removed the extra part and made a new slide before it going more high level.
